src/services/item/plugins/app/appData/service.ts

```diff
@@ -127,9 +151,13 @@ export class AppDataService {
       throw new PreventUpdateOtherAppData(appDataId);
     }
 
-    await this.hooks.runPreHooks('patch', member, repositories, appData);
```

Comment:
What is the reason for moving the pre hook before the above checks?

Reply:
I think this is a mistake. The checks should be performed before this hook.